src/shared/labels_op.c

```diff
@@ -94,85 +95,85 @@ int labels_format(const wlabel_t *labels, char *str, size_t size) {
 }
 
 /*
- * Parse labels from agent-info file.
+ * Parse labels from Wazuh DB - global.db - labels table.
  * Returns pointer to new null-key terminated array.
- * If no such file, returns NULL.
+ * If there are no labels for the agent, returns NULL.
  * Free resources with labels_free().
  */
-wlabel_t* labels_parse(const char *path) {
-    char buffer[OS_MAXSTR];
-    char *key;
-    char *value;
-    char *end;
-    size_t size = 0;
+wlabel_t* labels_parse(int agent_id) {
+    cJSON *json_labels = NULL;
+    cJSON *json_row_it = NULL;
+    char * str_key = NULL;
+    char * str_value = NULL;
+    char * key_start = NULL;
+    char * key_end = NULL;
     wlabel_t *labels;
     label_flags_t flags;
-    FILE *fp;
+    size_t size = 0;
 
-    if (!(fp = fopen(path, "r"))) {
-        if (errno == ENOENT) {
-            mdebug1(FOPEN_ERROR, path, errno, strerror(errno));
-        } else {
-            merror(FOPEN_ERROR, path, errno, strerror(errno));
-        }
+    json_labels = wdb_get_agent_labels(agent_id);
 
-        return NULL;
-    }
+    if (json_labels && json_labels->child) {
+        os_calloc(1, sizeof(wlabel_t), labels);
+
+        /*
+        "key1":value1\n
+        !"key2":value2\n
+        #"key3":_value3\n
+        */
+
+        os_calloc(OS_BUFFER_SIZE, sizeof(char), str_key);
+        os_calloc(OS_BUFFER_SIZE, sizeof(char), str_value);
+
+        cJSON_ArrayForEach (json_row_it, json_labels) {
+            key_start = NULL;
+            key_end = NULL;
+            os_strdup(cJSON_GetObjectItem(json_row_it, "key")->valuestring, str_key);
+            os_strdup(cJSON_GetObjectItem(json_row_it, "value")->valuestring, str_value);
```

Comment:
Shouldn't we check the string before using it?
`cJSON_IsString() && ()->valuestring != NULL`

Reply:
Or we may use `cJSON_GetObjectItemCaseSensitive()` ?
The documentation in [cJSON](https://github.com/DaveGamble/cJSON) says
`
Note that there are no NULL checks except for the result of cJSON_Parse because cJSON_GetObjectItemCaseSensitive checks for NULL inputs already, so a NULL value is just propagated and cJSON_IsNumber and cJSON_IsString return 0 if the input is NULL.`

Reply:
DONE